workflow/controller/controller.go

```diff
@@ -569,7 +581,15 @@ func (wfc *WorkflowController) processNextPodCleanupItem(ctx context.Context) bo
 	return true
 }
 
-func (wfc *WorkflowController) getPod(namespace string, podName string) (*apiv1.Pod, error) {
+func (wfc *WorkflowController) getPodFromAPI(ctx context.Context, namespace string, podName string) (*apiv1.Pod, error) {
+	pod, err := wfc.kubeclientset.CoreV1().Pods(namespace).Get(ctx, podName, metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+	return pod, nil
+}
```

Comment:
Did we address the issue from @isubasinghe [here](https://github.com/argoproj/argo-workflows/pull/12413#discussion_r1455463263)? What happens when there is a transient error such as one due to rate limiting? If we return an error, do we get a chance to try again later or no? 
@isubasinghe does the original code truly address this type of transient error anyway? If I look at the `labelPodCompleted` line in the original code, if it fails to run Patch(), it returns an error.

Reply:
ahh, okay, thanks. Seems good to me. Let me know if you see any issue @isubasinghe

Reply:
Okay, I suppose the main question is whether the [isTransient](https://github.com/argoproj/argo-workflows/blob/main/util/errors/errors.go#L31) method includes rate limiting. Maybe it doesn't. But perhaps that was a problem in the old code anyway.

Reply:
I think this: https://github.com/argoproj/argo-workflows/blob/main/util/errors/errors.go#L32 should handle rate limiting right?

Reply:
The new error handling addresses this but I think this must've meant that this was an issue in the old code as well. 
Do you think its a good idea to open a new issue for that and us make a note there that this PR will fix it? 

Or do you think a separate PR itself is more appropriate? 

Kinda happy to go with the first option but do admit that the second is cleaner.

Reply:
Ah no, this is my bad sorry, I for some reason didn't realise that the old code handled this as well. I sort of missed that it was inside a lambda. 


We can resolve this discussion I think.